src/common/src/array/stream_chunk.rs

```diff
@@ -181,12 +185,20 @@ impl StreamChunk {
     }
 
     pub fn from_parts(ops: Vec<Op>, data_chunk: DataChunk) -> Self {
-        let (columns, visibility) = data_chunk.into_parts();
+        let (columns, vis) = data_chunk.into_parts();
+        let visibility = match vis {
+            Vis::Bitmap(b) => Some(b),
+            Vis::Compact(_) => None,
+        };
         Self::new(ops, columns, visibility)
     }
 
     pub fn into_inner(self) -> (Vec<Op>, Vec<Column>, Option<Bitmap>) {
-        let (columns, visibility) = self.data.into_parts();
+        let (columns, vis) = self.data.into_parts();
+        let visibility = match vis {
+            Vis::Bitmap(b) => Some(b),
+            Vis::Compact(_) => None,
+        };
```

Comment:
It may feel a little bit redundant here and `impl From<Vis> for Option<Bitmap>` can make it shorter. However, I would like to make the caller of this conversion aware that cardinality info is discarded.

It is okay to throw away cardinality here because for StreamChunk we also have `ops.len()`. It is also okay below for `HashAgg::get_unique_keys` with a different reason, that cardinality info is in `keys` and `hash_codes`.

As of now these are the only cases. And I expect it to be rare.